learn/advanced/storage.md

```diff
@@ -71,3 +71,9 @@ These metrics are highly dependent on the machine that is running Meilisearch. R
 It is important to note that **there is no reliable way to predict the final size of a database**. This is true for just about any search engine on the market—we're just the only ones saying it out loud.
 
 Database size is affected by a large number of criteria, including settings, relevancy rules, use of facets, the number of different languages present, and more.
+
+## Soft deletion
+
+Meilisearch renders deleted documents inaccessible to all users but does not immediately remove them from the database. This is a common optimization technique called "soft deletion". Soft deleted documents are permanently deleted at a later date—the precise time depends on index size and available disk space. It might be important to check how soft deletion interacts with relevant data retention legislation.
```

Comment:
```suggestion
Meilisearch renders deleted documents inaccessible to all users but does not immediately remove them from the database. This is a common optimization technique called "soft deletion". Soft deleted documents are permanently deleted at a later date—the precise time depends on index size and available disk space.
```

I can see why it could interact with data retention legislation, but I find it strange to say it here while we don't talk about it at all when we explain the;
- a snapshot that stores the whole database in a big file
- a dump that stores the whole database on disk
- tasks that are slow to execute. The document deletions are not prioritized and thus may take time to execute
- Or a mix of both; document deletions that weren't executed and then got dumped/snapshotted will still contain all the data obviously.

But I'm not an expert at all in this domain.
As a developer, I think I would prefer seeing this information on another page about « legal information » to something instead of in the explanation of a feature :thinking: 

What do you think?

Reply:
Yeah sure :+1: